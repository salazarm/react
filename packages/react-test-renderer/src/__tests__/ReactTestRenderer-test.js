/**
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @emails react-core
 */

'use strict';

const React = require('react');
const ReactDOM = require('react-dom');

// Isolate test renderer.
jest.resetModules();
const ReactTestRenderer = require('react-test-renderer');

describe('ReactTestRenderer', () => {
  it('should support ReactDOM portal usage', () => {
    const container = document.createElement('div');
    let rendered = ReactTestRenderer.create(
      <div>
        {ReactDOM.createPortal(<span>Rendered by ReactDOM</span>, container)}
      </div>,
    );
    expect(rendered.toJSON()).toMatchSnapshot();

    rendered.update(
      <div>
        <span>Rendered by ReactTestRenderer</span>
        {ReactDOM.createPortal(<span>Rendered by ReactDOM</span>, container)}
      </div>,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
