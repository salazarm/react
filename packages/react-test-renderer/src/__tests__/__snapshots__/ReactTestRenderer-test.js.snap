// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactTestRenderer should support ReactDOM portal usage 1`] = `
<div>
  Object {
    "children": Array [
      Object {
        "children": Array [
          "Rendered by ReactDOM",
        ],
        "props": Object {},
        "type": "Symbol(react.element)",
      },
    ],
    "props": Object {},
    "type": "Symbol(react.portal)",
  }
</div>
`;

exports[`ReactTestRenderer should support ReactDOM portal usage 2`] = `
<div>
  Object {
    "children": Array [
      "Rendered by ReactTestRenderer",
    ],
    "props": Object {},
    "type": "Symbol(react.element)",
  }
  Object {
    "children": Array [
      Object {
        "children": Array [
          "Rendered by ReactDOM",
        ],
        "props": Object {},
        "type": "Symbol(react.element)",
      },
    ],
    "props": Object {},
    "type": "Symbol(react.portal)",
  }
</div>
`;
